bundle/src/main/java/com/adobe/acs/commons/properties/util/TemplateReplacementUtil.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.properties.util;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import org.apache.commons.lang.StringUtils;
+
+public class TemplateReplacementUtil {
```

Comment:
Any reason that this class isn't using a regex with a capture group for the key?

Reply:
I don't think there's a real reason for it to _not_ use one for getting the key, but there are util methods in this that are additional checks to avoid unnecessary processing (i.e. `hasPlaceholder` checks for the presence of the placeholders and the order of them). if this is a desired change i can make it

Reply:
I believe `hasPlaceholder()` and `getPlaceholder()` might fail in the following scenario: 

"Hello :-}}, how are you on this fine {{day of week}}?"

Likely not very common, but potentially possible on large bits of text.

Reply:
Yeah it will - that's actually why I have the JSON broken down to elements instead of string replacement too. I will update this to be a regex capture instead